src/memo/memo.mli

```diff
@@ -126,6 +111,20 @@ module Stack_frame : sig
   (** Checks if the stack frame is a frame of the given memoized function and if
       so, returns [Some i] where [i] is the argument of the function. *)
   val as_instance_of : t -> of_:('input, _) memo -> 'input option
+
+  val human_readable_description : t -> User_message.Style.t Pp.t option
+end
+
+module Error : sig
```

Comment:
The name `Error` seems too generic. My understanding is that this `Error` corresponds to errors raised by user-supplied functions that have been augmented with Memo stack information. If that's the case, perhaps we should call this module `User_error`, and also add the following as a comment:

```suggestion
(** Errors raised by user-supplied memoized functions that have been augmented with
    Memo call stack information. *)
module Error : sig
```

Reply:
The comment seems good. I'm not too sure about the name `User_error` because it is already used in Dune to mean "error that are reported in a way that the user can understand and act on".

Reply:
The word "user" considered too generic :-)

Reply:
Thinking a bit more, `User_error` is also too generic, since `Cycle_error` is a user error too, so I retract my suggestion. I'm still not quite happy about `Error` (because it doesn't include `Cycle_error` despite its generic name), but I can live with it.

Reply:
What about representing `Cycle_error` as a constructor of `Error`?

Reply:
Ah, yes, that sounds good actually!